client/state/happychat/actions.js

```diff
@@ -79,35 +81,62 @@ export const requestTranscript = () => ( dispatch, getState ) => {
 	);
 };
 
+const getSettings = new Promise( resolve => {
+	if ( userSettings.initialized ) {
+		return resolve( userSettings );
+	}
+	userSettings.once( 'change', () => {
+		resolve( userSettings );
+	} );
+} );
+
+const DEFAULT_LOCALE = 'en';
+
+const getLocale = getSettings.then( settings => {
+	const olarkState = olarkStore.get();
+	const olarkLocale = olarkState.locale;
+	const accountLocale = settings.getSetting( 'language' );
+
+	if ( accountLocale === DEFAULT_LOCALE ) {
+		debug( 'using olark chat locale' );
+		return olarkLocale;
+	}
+
+	return accountLocale;
+} );
```

Comment:
We could replace `userSettings` with a Redux selector:
```
import { getCurrentUserLocale } from 'state/current-user/selectors';

const locale = getCurrentUserLocale( state ) || 'en';
```

I'm also curious about the fallback to the olark locale. Is there a reason we couldn't just use `en` if that's the account locale?

It would be good if we can do away with the olark specific bits here, and currently the olark locale is basically the user's locale but limited to `es`, `pt-br` and `en`. If the user's locale is one we don't provide chat support for, we fall back to `en`. Say the user's locale was `ja` - in this case we display a message (in Japanese) on the contact form saying that support is only provided in English, but the user can still start an `en` chat.

Would the happychat service handle that fallback, if we pass `ja` as the locale? Or should we handle that fallback on the client side? If the latter, I'm tempted to say we just duplicate the list in Calypso for the time being, to simplify olark removal later.